spec/util/app-service-mock.js

```diff
@@ -0,0 +1,58 @@
+"use strict";
+var EventEmitter = require("events");
+var util = require("util");
+var instance = null;
+
+function MockAppService() {
+    EventEmitter.call(this);
+}
+util.inherits(MockAppService, EventEmitter);
+
+MockAppService.prototype.listen = function(port) {
+    // NOP
+};
+
+MockAppService.prototype._trigger = function(eventType, content) {
+    var promises = [];
+    var listeners = instance.listeners(eventType);
+    listeners.forEach(function(l) {
+        promises.push(l(content));
+    })
+    if (promises.length === 1) {
+        return promises[0];
+    }
+    return Promise.all(promises);
+};
+
+MockAppService.prototype._queryAlias = function(alias) {
+    if (!this.onAliasQuery) {
+        throw new Error("IRC AS hasn't hooked into onAliasQuery yet.");
+    }
+    return this.onAliasQuery(alias);
+};
+
+MockAppService.prototype._queryUser = function(user) {
+    if (!this.onUserQuery) {
+        throw new Error("IRC AS hasn't hooked into onUserQuery yet.");
+    }
+    return this.onUserQuery(user);
+};
+
+function MockAppServiceProxy() {
+    if (!instance) {
+        instance = new MockAppService();
+    }
+    return instance;
+}
+MockAppServiceProxy.instance = function() {
```

Comment:
This exists because we have no DI framework and need to be referring to the same instance of `AppService` as the one that is `new`'d by the bridge. This is how we do that (by using an `instance` singleton).